cogs/images.py

```diff
@@ -66,97 +66,89 @@ def _shifter(self, attachment_bytes: bytes, size: tuple):
 
         new_image = Image.merge('RGB', (new_red, new_green, new_blue))
 
-        output = BytesIO()
-        new_image.save(output, format="png")
-        output.seek(0)
-        
-        return output
+        out_file = BytesIO()
+        new_image.save(out_file)
+        out_file.seek(0)
+        return out_file
 
+    def _jpeg(self, attachment_file: BytesIO, severity: int) -> BytesIO:
+        image_obj = Image.open(attachment_file)
 
-    async def _get_image(self, ctx: NewCtx) -> tuple:
-        if not ctx.message.attachments:
-            attachment_bytes = await ctx.author.avatar_url_as(size=1024, format='jpg').read()
-            filename = ctx.author.display_name + '.jpg'
-            filesize = (1024, 1024)
-        else:
-            target = ctx.message.attachments[0]
-            attachment_bytes = await target.read()
-            filename = target.filename
-            filesize = (target.width, target.height)
+        out_file = BytesIO()
+        image_obj.save(out_file, format='JPEG', quality=severity)
+        out_file.seek(0)
+        return out_file
 
-        return attachment_bytes, filename, filesize
+    def _loop_jpeg(self, attachment_file: BytesIO, severity: int, loops: int) -> BytesIO:
+        for _ in range(loops):
+            attachment_file = self._jpeg(attachment_file, severity)
+        return attachment_file
 
-    def loop_jpeg(self, severity, filename, loopyloops):
-        for _ in range(loopyloops):
-            image = Image.open(filename)
-            image.save(filename, format='jpeg', quality=severity)
-            image.close()
+    def _get_image(self, ctx: NewCtx) -> Tuple[BytesIO, str, Tuple[int, int]]:
+        attachment_file = BytesIO()
 
+        if not ctx.message.attachments:
+            await ctx.author.avatar_url_as(size=1024, format='png').save(attachment_file)
```

Comment:
what are the ytes being returned to?

Reply:
they're written to `attachment_file` the BytesIO object

Reply:
woh,sorry, you right